### PR TITLE
slow-query: add `Warnings` field to slow log and the slow query table

### PR DESCRIPTION
**Pull request by time-and-fate**
### What is changed, added or deleted? (Required)

Starting from TiDB v6.6.0, the `Warnings` field is added in slow query log

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
  https://github.com/pingcap/tidb/issues/39893
  https://github.com/pingcap/tidb/pull/39884

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: identify-slow-queries.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ Slow Query 基础信息：
 * `Backoff_total`：表示这个语句在执行过程中所有 backoff 花费的时间。
 * `Write_sql_response_total`：表示这个语句把结果发送回客户端花费的时间。
 * `Result_rows`：表示这个语句查询结果的行数。
+* `Warnings`: 以 JSON 格式表示的该语句执行过程中产生的警告。内容通常和 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 语句的输出结果一致。但是可能会包含 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 中不会出现的警告来提供更多诊断信息，这类警告会标注 `IsExtra: true`。
```
Comment (Oreoxmt) [inline identify-slow-queries.md:85]:
```suggestion
* `Warnings`：以 JSON 格式表示的该语句执行过程中产生的警告。通常和 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 语句的输出结果一致，但是可能会包含 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 中没有出现的警告来提供更多诊断信息。这类警告将被标记为 `IsExtra: true`。
```

**Comment by Oreoxmt** [comment]
@time-and-fate Could you please invite a tech reviewer? Thanks!

**Comment by Oreoxmt** [comment]
@xuyifangreeneyes PTAL

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: identify-slow-queries.md [hunk 2]
```diff
@@ -82,6 +82,7 @@ Slow Query 基础信息：
 * `Backoff_total`：表示这个语句在执行过程中所有 backoff 花费的时间。
 * `Write_sql_response_total`：表示这个语句把结果发送回客户端花费的时间。
 * `Result_rows`：表示这个语句查询结果的行数。
+* `Warnings`：以 JSON 格式表示的该语句执行过程中产生的警告。通常和 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 语句的输出结果一致，但是可能会包含 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 中没有出现的警告来提供更多诊断信息。这类警告将被标记为 `IsExtra: true`。
```
Comment (shichun-0415) [inline identify-slow-queries.md:85]:
```suggestion
* `Warnings`：表示这个语句执行过程中产生的警告，采用 JSON 格式。通常和 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 语句的输出结果一致，但是可能会包含 [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) 中没有出现的警告来提供更多诊断信息。这类警告将被标记为 `IsExtra: true`。
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

**Pull request merged**